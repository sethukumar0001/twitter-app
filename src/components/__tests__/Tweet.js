import React from 'react';
import renderer from 'react-test-renderer';
import Tweet from '../Tweet';

jest.mock('moment', () => time => ({
  fromNow: () => `${time} ago`,
}));

jest.mock('react-router-dom');

describe('Tweet', () => {
  const tweet = {
    id: '123',
    text: 'Hello, world!',
    createdAt: '1',
    user: {
      id: '321',
      username: 'user2',
    },
  };

  const reply = {
    id: '456',
    text: 'Hello!',
    createdAt: '2',
    user: {
      id: '654',
      username: 'user2',
    },
  };

  it('renders correctly', () => {
    const component = renderer.create(<Tweet {...tweet} />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders highlighted tweet', () => {
    const component = renderer.create(<Tweet {...tweet} highlighted />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders reply', () => {
    const component = renderer.create(
      <Tweet {...reply} replyToId={tweet.id} repliedTweet={tweet} />,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
