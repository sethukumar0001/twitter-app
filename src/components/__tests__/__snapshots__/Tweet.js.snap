// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tweet renders correctly 1`] = `
<li
  className="MuiPaper-root-6 MuiPaper-elevation1-9 MuiPaper-rounded-7 MuiCard-root-5 Tweet-card-1"
>
  <div
    className="MuiCardHeader-root-33"
  >
    <div
      className="MuiCardHeader-avatar-34"
    >
      <div
        className="MuiAvatar-root-39 MuiAvatar-colorDefault-40"
        style={
          Object {
            "backgroundColor": "#3f51b5",
          }
        }
      >
        u
      </div>
    </div>
    <div
      className="MuiCardHeader-content-36"
    >
      <span
        className="MuiTypography-root-42 MuiTypography-body2-50 MuiCardHeader-title-37"
      >
        user2
      </span>
      <span
        className="MuiTypography-root-42 MuiTypography-body2-50 MuiTypography-colorTextSecondary-65 MuiCardHeader-subheader-38"
      />
    </div>
  </div>
  <div
    className="MuiCardContent-root-67 Tweet-content-3"
  >
    <h3
      className="MuiTypography-root-42 MuiTypography-subheading-49"
    >
      <div>
        <p>
          Hello, world!
        </p>
      </div>
    </h3>
  </div>
  <div
    className="MuiCardActions-root-68"
  />
</li>
`;

exports[`Tweet renders highlighted tweet 1`] = `
<div
  className="MuiPaper-root-6 MuiPaper-elevation8-16 MuiPaper-rounded-7 MuiCard-root-5 Tweet-card-1"
>
  <div
    className="MuiCardHeader-root-33"
  >
    <div
      className="MuiCardHeader-avatar-34"
    >
      <div
        className="MuiAvatar-root-39 MuiAvatar-colorDefault-40"
        style={
          Object {
            "backgroundColor": "#3f51b5",
          }
        }
      >
        u
      </div>
    </div>
    <div
      className="MuiCardHeader-content-36"
    >
      <span
        className="MuiTypography-root-42 MuiTypography-body2-50 MuiCardHeader-title-37"
      >
        user2
      </span>
      <span
        className="MuiTypography-root-42 MuiTypography-body2-50 MuiTypography-colorTextSecondary-65 MuiCardHeader-subheader-38"
      />
    </div>
  </div>
  <div
    className="MuiCardContent-root-67 Tweet-content-3"
  >
    <h1
      className="MuiTypography-root-42 MuiTypography-display1-46"
    >
      <div>
        <p>
          Hello, world!
        </p>
      </div>
    </h1>
  </div>
</div>
`;

exports[`Tweet renders reply 1`] = `
<li
  className="MuiPaper-root-6 MuiPaper-elevation1-9 MuiPaper-rounded-7 MuiCard-root-5 Tweet-card-1"
>
  <div
    className="MuiCardHeader-root-33"
  >
    <div
      className="MuiCardHeader-avatar-34"
    >
      <div
        className="MuiAvatar-root-39 MuiAvatar-colorDefault-40"
        style={
          Object {
            "backgroundColor": "#3f51b5",
          }
        }
      >
        u
      </div>
    </div>
    <div
      className="MuiCardHeader-content-36"
    >
      <span
        className="MuiTypography-root-42 MuiTypography-body2-50 MuiCardHeader-title-37"
      >
        user2
      </span>
      <span
        className="MuiTypography-root-42 MuiTypography-body2-50 MuiTypography-colorTextSecondary-65 MuiCardHeader-subheader-38"
      />
    </div>
  </div>
  <div
    className="MuiCardContent-root-67 Tweet-content-3"
  >
    <span
      className="MuiTypography-root-42 MuiTypography-caption-52"
    >
      In reply to
       
    </span>
    <h3
      className="MuiTypography-root-42 MuiTypography-subheading-49"
    >
      <div>
        <p>
          Hello!
        </p>
      </div>
    </h3>
  </div>
  <div
    className="MuiCardActions-root-68"
  />
</li>
`;
